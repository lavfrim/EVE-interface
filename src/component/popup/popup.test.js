import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import PopUp from './index';

function Component() {
    return (
        <p>
            {'smth'}
        </p>
    );
}

describe('PopUp should render correctly', () => {
    it('with stage = basic', () => {
        const wrapper = mount(
            <PopUp stage={0}>
                <Component />
            </PopUp>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = clicked', () => {
        const wrapper = mount(
            <PopUp stage={1}>
                <Component />
            </PopUp>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = returned', () => {
        const wrapper = mount(
            <PopUp stage={-1}>
                <Component />
            </PopUp>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

