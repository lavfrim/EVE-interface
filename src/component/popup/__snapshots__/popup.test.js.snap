// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp should render correctly with stage = basic 1`] = `
<PopUp
  stage={0}
>
  <div
    className="pop-up"
    id="outside"
    onClick={[Function]}
  >
    <div
      className="pop-up__content-container"
    >
      <button
        className="pop-up__content-container__close-button"
        id="close"
      >
        X
      </button>
      <div
        className="pop-up__content-container__content"
      >
        <Component>
          <p>
            smth
          </p>
        </Component>
      </div>
    </div>
  </div>
</PopUp>
`;

exports[`PopUp should render correctly with stage = clicked 1`] = `
<PopUp
  stage={1}
>
  <div
    className="pop-up"
    id="outside"
    onClick={[Function]}
  >
    <div
      className="pop-up__content-container"
    >
      <button
        className="pop-up__content-container__close-button"
        id="close"
      >
        X
      </button>
      <div
        className="pop-up__content-container__content"
      >
        <Component>
          <p>
            smth
          </p>
        </Component>
      </div>
      <button
        className="pop-up__content-container__nav-button"
        id="backward"
      >
        &lt;
      </button>
    </div>
  </div>
</PopUp>
`;

exports[`PopUp should render correctly with stage = returned 1`] = `
<PopUp
  stage={-1}
>
  <div
    className="pop-up"
    id="outside"
    onClick={[Function]}
  >
    <div
      className="pop-up__content-container"
    >
      <button
        className="pop-up__content-container__close-button"
        id="close"
      >
        X
      </button>
      <div
        className="pop-up__content-container__content"
      >
        <Component>
          <p>
            smth
          </p>
        </Component>
      </div>
      <button
        className="pop-up__content-container__nav-button"
        id="forward"
      >
        &gt;
      </button>
    </div>
  </div>
</PopUp>
`;
